src/Relations/BelongsToThrough.php

```diff
@@ -385,15 +351,20 @@ public function getFirstForeignKeyName()
      */
     public function getQualifiedFirstLocalKeyName()
     {
-        /** @var TIntermediateModel $lastThroughParent */
         $lastThroughParent = end($this->throughParents);
 
+        if ($lastThroughParent === false) {
```

Comment:
@staudenmeir when `false` is returned, what model should we fall back to? Inside `getForeignKeyName` we fall back to `$this->parent`, so for now I've updated this code to do the same

Reply:
That should never happen? Maybe just throw an exception unless there's a valid case for it

Reply:
Good idea